app/backend/umldiagram/umldiagram_test.go

```diff
@@ -418,6 +413,273 @@ func TestAddAttributeToGadget(t *testing.T) {
 	assert.Equal(t, "can only operate on one component", err.Error())
 }
 
+func TestLoadExistUMLDiagram(t *testing.T) {
+
+}
+
+func TestLoadGadgetAttributes(t *testing.T) {
+	dia, err := CreateEmptyUMLDiagram("TestDiagram", ClassDiagram)
+	assert.NoError(t, err)
+	assert.NotNil(t, dia)
+
+	expectedContent := "test content"
+	expectedSize := 12
+	expectedStyle := attribute.Textstyle(attribute.Bold | attribute.Italic)
+	expectedFontFile := os.Getenv("APP_ROOT") + "/frontend/src/assets/fonts/Inkfree.ttf"
+
+	savedAttributeBase := utils.SavedAtt{
+		Content:  expectedContent,
+		Size:     expectedSize,
+		Style:    int(expectedStyle),
+		FontFile: expectedFontFile,
+	}
+	savedAttributes := make([]utils.SavedAtt, 3)
+	for i := 0; i < 3; i++ {
+		savedAttributes[i] = savedAttributeBase
+		savedAttributes[i].Size += i
+		savedAttributes[i].Ratio = 0.3 * float64(i)
+	}
+
+	gad, err := component.NewGadget(component.Class, utils.Point{}, 0, "someInvalidColorHex", "")
+	assert.NoError(t, err)
+	assert.NotNil(t, gad)
+
+	err, _ = dia.loadGadgetAttributes(gad, savedAttributes) // Err-index is not important cuz we expect err is nil
+	assert.NoError(t, err)
+
+	loadedAttributes := gad.GetAttributes()
+
+	for i := 0; i < 3; i++ {
+		assert.Equal(t, 1, len(loadedAttributes[i]))
+		assert.Equal(t, savedAttributes[i].Content, loadedAttributes[i][0].GetContent())
+		assert.Equal(t, savedAttributes[i].Style, int(loadedAttributes[i][0].GetStyle()))
+		assert.Equal(t, savedAttributes[i].FontFile, loadedAttributes[i][0].GetFontFile())
+		assert.Equal(t, savedAttributes[i].Size, loadedAttributes[i][0].GetSize())
+		assert.Equal(t, savedAttributes[i].FontFile, loadedAttributes[i][0].GetFontFile())
+	}
+}
+func TestLoadGadgetAttributesButWithJsonStr(t *testing.T) {
+	dia, err := CreateEmptyUMLDiagram("TestDiagram", ClassDiagram)
+	assert.NoError(t, err)
+	assert.NotNil(t, dia)
+
+	jsonStr := `{
+		"Content": "test content",
+		"Size": 12,
+		"Style": 3,
+		"FontFile": "` + os.Getenv("APP_ROOT") + `/frontend/src//assets/fonts/Inkfree.ttf"
+	}`
+
```

Comment:
```suggestion
	jsonStr = strings.ReplaceAll(jsonStr, "\\", "/")

```
replace "\" into "/" otherwise this test fails on windown occasionally.